sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/ProtobufDataImporter.java

```diff
@@ -47,15 +55,42 @@ public ProtobufDataImporter(FileLinesContextFactory fileLinesContextFactory, NoS
     this.noSonarFilter = noSonarFilter;
   }
 
-  public void importResults(SensorContext context, Path protobufReportsDirectory, String repositoryKey, boolean importIssues) {
-    // Note: the no-sonar "measure" must be imported before issues, otherwise the affected issues won't get excluded!
-    parseProtobuf(ProtobufImporters.metricsImporter(context, fileLinesContextFactory, noSonarFilter), protobufReportsDirectory, METRICS_OUTPUT_PROTOBUF_NAME);
-    if (importIssues) {
-      parseProtobuf(ProtobufImporters.issuesImporter(context, repositoryKey), protobufReportsDirectory, ISSUES_OUTPUT_PROTOBUF_NAME);
+  public void importResults(SensorContext context, List<Path> protobufReportsDirectories, String repositoryKey, boolean importIssues) {
+    RawProtobufImporter<MetricsInfo> metricsImporter = ProtobufImporters.metricsImporter(context, fileLinesContextFactory, noSonarFilter);
+    RawProtobufImporter<FileIssues> issuesImporter = ProtobufImporters.issuesImporter(context, repositoryKey);
+    RawProtobufImporter<TokenTypeInfo> highlightImporter = ProtobufImporters.highlightImporter(context);
+    RawProtobufImporter<SymbolReferenceInfo> symbolRefsImporter = ProtobufImporters.symbolRefsImporter(context);
+    RawProtobufImporter<CopyPasteTokenInfo> cpdTokensImporter = ProtobufImporters.cpdTokensImporter(context);
+
+    for (Path protobufReportsDir : protobufReportsDirectories) {
+      long protoFiles = countProtoFiles(protobufReportsDir);
+      LOG.info(String.format("Importing results from %d proto %s in '%s'", protoFiles, pluralize("file", protoFiles), protobufReportsDir));
+      // Note: the no-sonar "measure" must be imported before issues, otherwise the affected issues won't get excluded!
+      parseProtobuf(metricsImporter, protobufReportsDir, METRICS_OUTPUT_PROTOBUF_NAME);
+      if (importIssues) {
+        parseProtobuf(issuesImporter, protobufReportsDir, ISSUES_OUTPUT_PROTOBUF_NAME);
+      }
+      parseProtobuf(highlightImporter, protobufReportsDir, HIGHLIGHT_OUTPUT_PROTOBUF_NAME);
+      parseProtobuf(symbolRefsImporter, protobufReportsDir, SYMBOLREFS_OUTPUT_PROTOBUF_NAME);
+      parseProtobuf(cpdTokensImporter, protobufReportsDir, CPDTOKENS_OUTPUT_PROTOBUF_NAME);
+    }
+  }
+
+  private static String pluralize(String s, long count) {
+    if (count == 1) {
+      return s;
+    }
+    return s + "s";
+  }
+
+  private static long countProtoFiles(Path dir) {
+    try {
+      return Files.list(dir)
+        .filter(p -> p.getFileName().toString().toLowerCase().endsWith(".pb"))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define the locale to be used in this String operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1449)